ui/e2e/explorer/workflow/instance.spec.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  createInstance,
+  simpleWorkflow as simpleWorkflowContent,
+} from "./utils";
+import { createNamespace, deleteNamespace } from "../../utils/namespace";
+import { expect, test } from "@playwright/test";
+
+import { createFile } from "e2e/utils/files";
+import { faker } from "@faker-js/faker";
+
+let namespace = "";
+const simpleWorkflowName = faker.system.commonFileName("yaml");
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+  /* create workflows we can use to create instances later */
+  await createFile({
+    name: simpleWorkflowName,
+    namespace,
+    type: "workflow",
+    yaml: simpleWorkflowContent,
+  });
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+  namespace = "";
+});
+
+test("the header of the instance page shows the relevant data for the workflow", async ({
+  page,
+}) => {
+  const newInstance = createInstance({ namespace, path: simpleWorkflowName });
+
+  await expect(newInstance, "wait until process was completed").toBeDefined();
+
+  const instanceID = (await newInstance).instance;
+
+  await page.goto(`/${namespace}/instances/${instanceID}`);
+
+  const header = page.getByTestId("instance-header-container");
+  await expect(header, "the header is visible").toBeVisible();
+
+  const instanceID_Header = header.locator("h3");
+  await expect(instanceID_Header, "the instance ID is visible").toBeVisible();
+
+  const instanceID_navLink = page.locator("ul").locator("a").nth(2);
+
+  expect(instanceID_Header.innerText, "the instance IDs are the same").toEqual(
+    instanceID_navLink.innerText
+  );
+
+  await expect(
+    header.locator("div").first(),
+    "the badge complete is visible"
+  ).toContainText("complete");
+
+  const trigger = header.locator('div.text-sm:has-text("trigger")');
```

Comment:
Some locators are very tricky to find and as a general rule of thumb, we should not use tailwind classes in our selectors or avoid classes as good as we can. 

My suggestion would be

```
 await expect(header.getByText("triggerapi"), "It renders the instance trigger").toBeVisible();
```

This has a little downside as well, it's not super readable because "triggerapi" looks a bit weird with no line break, but I found it actually very handy because it somehow also tests that the label (trigger) is rendered next to (api).

Reply:
Yeah, a bit weird, but I will continue with this schema. 👍